pandas/tests/io/excel/test_readers.py

```diff
@@ -164,6 +164,35 @@ def xfail_datetimes_with_pyxlsb(engine, request):
 
 
 class TestReaders:
+    @pytest.mark.parametrize("col", [[True, None, False], [True], [True, False]])
+    def test_read_excel_type_check(self, col):
+        # GH 58159
+        df = DataFrame({"bool_column": col}, dtype=pd.BooleanDtype.name)
```

Comment:
```suggestion
        df = DataFrame({"bool_column": col}, dtype="boolean")
```

And could you change other usages here?